Tests/ConvertTo-BMOtterScriptExpression.Tests.ps1

```diff
@@ -0,0 +1,110 @@
+Set-StrictMode -Version 'Latest'
+
+BeforeAll {
+    Set-StrictMode -Version 'Latest'
+
+    & (Join-Path -Path $PSScriptRoot -ChildPath 'Initialize-Tests.ps1' -Resolve)
+
+    function GivenValue
+    {
+        param(
+            $Value
+        )
+
+        $script:value = $Value
+    }
+
+    function WhenConverting
+    {
+        $warnings = @()
+        $script:result = ConvertTo-BMOtterScriptExpression -Value $script:value -WarningVariable 'warnings'
+        $script:warnings = $warnings
+    }
+
+    function ThenIsJson
+    {
+        $script:result | ConvertFrom-Json
+    }
+
+    function ThenIsMap
+    {
+        $script:result.StartsWith('%(') | Should -BeTrue
+        $script:result.EndsWith(')') | Should -BeTrue
+    }
+
+    function ThenIsVector
+    {
+        $script:result.StartsWith('@(') | Should -BeTrue
+        $script:result.EndsWith(')') | Should -BeTrue
+    }
+
+    function ThenEquals
+    {
+        param(
+            $Value
+        )
+
+        $script:result | Should -Be $Value
+    }
+
+    function ThenWarn
+    {
+        param(
+            $Message
+        )
+
+        $script:warnings | Should -Not -BeNullOrEmpty
+        $script:warnings | Should -HaveCount 1
+        $script:warnings | Select-Object -First 1 | Should -BeLike $Message
+    }
+}
+
+Describe 'ConvertTo-BMOtterScriptExpression' {
+    BeforeEach {
+        $script:value = $null
+        $script:result = $null
+        $script:warnings = $null
+    }
+
+    It 'should result in a string map' {

```

Comment:
Add a test for when an object that isn't a String, Array, or Hashtable is passed in.